urllib3/util/retry.py

```diff
@@ -194,12 +206,51 @@ def get_backoff_time(self):
         backoff_value = self.backoff_factor * (2 ** (consecutive_errors_len - 1))
         return min(self.BACKOFF_MAX, backoff_value)
 
-    def sleep(self):
-        """ Sleep between retry attempts using an exponential backoff.
+    def parse_retry_after(self, retry_after):
+        if retry_after.isdigit():
+            seconds = int(retry_after)
+        else:
+            retry_date_tuple = email.utils.parsedate(retry_after)
+            if retry_date_tuple is None:
+                raise ValueError("Invalid Retry-After header: %s" % retry_after)
```

Comment:
@haikuginger Yes, that's why I used `ValueError`. It would be better if we have an exception for invalid HTTP header. I'll find better one or create new one.